app/src/main/java/emu/skyline/applet/swkbd/SoftwareKeyboardConfig.kt

```diff
@@ -0,0 +1,238 @@
+/*
+ * SPDX-License-Identifier: MPL-2.0
+ * Copyright © 2022 Skyline Team and Contributors (https://github.com/skyline-emu/)
+ */
+
+@file:OptIn(ExperimentalUnsignedTypes::class)
+
+package emu.skyline.applet.swkbd
+
+import emu.skyline.utils.*
+
+val DownloadCodeCodepoints = intArrayOf('0'.code, '1'.code, '2'.code, '3'.code, '4'.code, '5'.code, '6'.code, '7'.code, '8'.code, '9'.code, 'A'.code, 'B'.code, 'C'.code, 'D'.code, 'E'.code, 'F'.code, 'G'.code, 'H'.code, 'J'.code, 'K'.code, 'L'.code, 'M'.code, 'N'.code, 'P'.code, 'Q'.code, 'R'.code, 'S'.code, 'T'.code, 'U'.code, 'V'.code, 'W'.code, 'X'.code)
+val OutsideOfMiiNicknameCodepoints = intArrayOf('@'.code, '%'.code, '\\'.code, 'ō'.code, 'Ō'.code, '₽'.code, '₩'.code, '♥'.code, '♡'.code)
+
+/**
+ * This data class matches KeyboardMode in skyline/applet/swkbd/software_keyboard_config.h
+ */
+data class KeyboardMode(
+    var mode : u32 = 0u
+) : ByteBufferSerializable {
+    companion object {
+        val Full = KeyboardMode(0u)
+        val Numeric = KeyboardMode(1u)
+        val ASCII = KeyboardMode(2u)
+        val FullLatin = KeyboardMode(3u)
+        val Alphabet = KeyboardMode(4u)
+        val SimplifiedChinese = KeyboardMode(5u)
+        val TraditionalChinese = KeyboardMode(6u)
+        val Korean = KeyboardMode(7u)
+        val LanguageSet2 = KeyboardMode(8u)
+        val LanguageSet2Latin = KeyboardMode(9u)
+    }
+}
+
+/**
+ * This data class matches InvalidCharFlags in skyline/applet/swkbd/software_keyboard_config.h
+ */
+data class InvalidCharFlags(
+    var flags : u32 = 0u
+) : ByteBufferSerializable {
+    // Access each flag as a bool
+    var space : Boolean
+        get() : Boolean = (flags and 0b1u shl 1) != 0u
```

Comment:
Move every getter in this class to this syntax (see below).

Reply:
Helo

Reply:
https://github.com/skyline-emu/skyline/pull/1616#discussion_r906787456